src/builder-context.c

```diff
@@ -368,7 +369,7 @@ builder_context_download_uri (BuilderContext *self,
   if (self->sources_urls != NULL)
     {
       g_autofree char *base_name = g_path_get_basename (soup_uri_get_path (original_uri));
-      g_autofree char *rel = g_build_filename ("downloads", sha256, base_name, NULL);
+      g_autofree char *rel = g_build_filename ("downloads", checksums[0], base_name, NULL);
```

Comment:
What’s special about index 0 here?

Reply:
The list is ordered such that the first is the "default" to use for when you only can use one checksum, such as in the filename in the download cache. See the comments in builder_get_all_checksums.